pillarbox-player/src/main/java/ch/srgssr/pillarbox/player/tracker/CurrentMediaItemTracker.kt

```diff
@@ -50,158 +46,134 @@ internal class CurrentMediaItemTracker internal constructor(
         set(value) {
             if (field == value) return
             field = value
-            if (field) {
-                currentMediaItem = player.currentMediaItem
-                if (currentMediaItem.canHaveTrackingSession()) {
-                    currentMediaItem?.let { startNewSession(it) }
-                }
-            } else {
-                trackers?.let { stopSession(MediaItemTracker.StopReason.Stop, player.currentPosition) }
-            }
+            setMediaItem(player.currentMediaItem)
         }
 
     private val window = Window()
 
     init {
-        player.addAnalyticsListener(this)
+        // player.addAnalyticsListener(this)
+        player.addListener(this)
         player.currentMediaItem?.let { startNewSession(it) }
     }
 
-    private fun stopSession(stopReason: MediaItemTracker.StopReason, positionMs: Long) {
-        if (currentMediaItem.canHaveTrackingSession()) {
-            trackers?.let {
-                for (tracker in it.list) {
-                    tracker.stop(player, stopReason, positionMs)
-                }
+    /**
+     * Set media item if has not tracking data, set to null
+     */
+    private fun setMediaItem(mediaItem: MediaItem?) {
+        if (enabled && mediaItem.canHaveTrackingSession()) {
+            if (!areEqual(currentMediaItem, mediaItem)) {
+                currentItemChange(currentMediaItem, mediaItem)
+                currentMediaItem = mediaItem
             }
-        }
-        trackers = null
-        this.currentMediaItem = null
-    }
-
-    private fun startNewSession(mediaItem: MediaItem?) {
-        currentMediaItem?.let { stopSession(MediaItemTracker.StopReason.Stop, player.currentPosition) }
-        currentMediaItem = mediaItem
-        if (enabled && mediaItem.isLoaded()) {
+        } else {
             currentMediaItem?.let {
-                startSessionInternal(it)
+                stopSession(MediaItemTracker.StopReason.Stop, player.currentPosition)
             }
         }
     }
 
-    private fun updateOrStartSession(mediaItem: MediaItem?) {
-        if (!enabled) {
+    private fun currentItemChange(lastMediaItem: MediaItem?, newMediaItem: MediaItem?) {
+        require(newMediaItem.canHaveTrackingSession())
+        if (newMediaItem == null) {
```

Comment:
If `newMediaItem` is `null`, `newMediaItem.canHaveTrackingSession()` will return `true`, and an exception will be thrown.
So the check here is not necessary.
Should it be done before the `require`?